repose-aggregator/components/filters/body-extractor-to-header/src/main/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.bodyextractortoheader
+
+import java.net.URL
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.jayway.jsonpath.{DocumentContext, JsonPath, Configuration => JsonConfiguration, Option => JsonOption}
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.MutableHttpServletRequest
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.bodyextractortoheader.config.BodyExtractorToHeaderConfig
+
+import scala.collection.JavaConverters._
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+@Named
+class BodyExtractorToHeaderFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[BodyExtractorToHeaderConfig] with LazyLogging {
+
+  import BodyExtractorToHeaderFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var extractions: Iterable[Extraction] = _
+  val jsonPathConfiguration = JsonConfiguration.defaultConfiguration()
+  jsonPathConfiguration.addOptions(JsonOption.DEFAULT_PATH_LEAF_TO_NULL)
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("Body Extractor to Header filter initializing...")
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info("Initializing Body Extractor to Header filter using config {}", configurationFile)
+    val xsdUrl: URL = getClass.getResource(SCHEMA_FILE_NAME)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFile, xsdUrl, this, classOf[BodyExtractorToHeaderConfig])
+
+    logger.trace("Body Extractor to Header filter initialized.")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val mutableHttpRequest = MutableHttpServletRequest.wrap(servletRequest.asInstanceOf[HttpServletRequest])
+
+    def addHeader(name: String, value: String, quality: Option[java.lang.Double], overwrite: Boolean): Unit = {
+      if (overwrite) {
+        mutableHttpRequest.removeHeader(name)
+      }
+      val hdrValue = quality match {
+        case Some(qual) => s"$value;q=$qual"
+        case None => value
+      }
+      mutableHttpRequest.addHeader(name, hdrValue)
+    }
+
+    val jsonDoc: Try[DocumentContext] = {
+      Option(mutableHttpRequest.getContentType) match {
+        case Some(contentType) =>
+          if (contentType.toLowerCase.contains("json")) {
+            val is = mutableHttpRequest.getInputStream
+            if (is.markSupported()) is.mark(0)
```

Comment:
I couldn't find it documented anywhere about `InputStream.mark(0)` being unlimited either, but it has always worked. That being said, I updated it to `Integer.MAX_VALUE` which is actually correct.